### PR TITLE
refine fullbackup

### PR DESCRIPTION
**Pull request by weekface**
### What is changed and how it works?

- get the TS first and increase the TiKV GC life time to 3 hours before the full backup, fixes: #560 
- add more options to mydumper
- upgrade mydumper image to pingcap/tidb-cloud-backup:20190610
- add an options to stability test

This pr passed the stability tests on cluster 1.

### Check List 

Tests 

 - E2E test
 - Stability test

Code changes

 - Has Helm charts change
 - Has Go code change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
get the TS first and increase the TiKV GC life time to 3 hours before the full backup
 ```

### CODE REVIEW

File: charts/tidb-backup/templates/scripts/_start_backup.sh.tpl [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
   --port=4000 \
   --user=${TIDB_USER} \
   --password=${TIDB_PASSWORD} \
   --tidb-snapshot=${savepoint} \
+  --long-query-guard=3600 \
+  --tidb-force-priority=LOW_PRIORITY \
   {{ .Values.backupOptions }}
 
+echo "Reset TiKV GC life time to 10m"
+/usr/bin/mysql -h${host} -P4000 -u${TIDB_USER} -p${TIDB_PASSWORD} -Nse "update mysql.tidb set variable_value='10m' where variable_name='tikv_gc_life_time';"
```
Comment (gregwebs) [inline charts/tidb-backup/templates/scripts/_start_backup.sh.tpl]: It looks like this assumes 10 minutes was previously set, but I think we need to get the original value and set it back to that original value.

File: charts/tidb-backup/templates/backup-job.yaml [hunk 2]
```diff
@@ -23,14 +23,10 @@ spec:
       - name: get-ts
         image: {{ .Values.image.binlog }}
         imagePullPolicy: {{ .Values.image.pullPolicy | default "IfNotPresent" }}
-        command:
-        - /binlogctl
-        - -pd-urls=http://{{ .Values.clusterName }}-pd:2379
-        - -cmd=generate_meta
-        - -data-dir=/savepoint-dir
+        command: ["/bin/sh", "-c", "cp /binlogctl /shared-dir/"]
```
Comment (weekface) [inline charts/tidb-backup/templates/backup-job.yaml]: The initContainer only copy binlogctl to the backup container

**Comment by cofyc** [comment]
```
failed to watch file "/var/log/pods/e804fae3-8df2-11e9-8cad-0242aae97541/tidb-operator-e2e/0.log": no space left on device++ kubectl get po tidb-operator-e2e -n tidb-operator-e2e '-ojsonpath={.status.phase}'
```
no enough space left?

**Comment by cofyc** [comment]
/run-e2e-tests
try again

File: charts/tidb-backup/templates/scripts/_start_backup.sh.tpl [hunk 3]
```diff
@@ -1,26 +1,46 @@
 set -euo pipefail
 
-host=`echo {{ .Values.clusterName }}_TIDB_SERVICE_HOST | tr '[a-z]' '[A-Z]' | tr '-' '_'`
+host_env=`echo {{ .Values.clusterName }}_TIDB_SERVICE_HOST | tr '[a-z]' '[A-Z]' | tr '-' '_'`
+host=`eval echo '${'$host_env'}'`
```
Comment (tennix) [inline charts/tidb-backup/templates/scripts/_start_backup.sh.tpl]:
```
host=$(getent hosts {{ .Values.clusterName }} | head | awk '{print $1}')
```

**Comment by weekface** [comment]
@tennix ptal

**Pull request merged**